common/src/autogluon/common/utils/utils.py

```diff
@@ -25,7 +25,7 @@
 
 
 def setup_outputdir(
-    path: str | None, warn_if_exist: bool = True, create_dir: bool = True, path_suffix: str | None = None
+    path: str | Path | None, warn_if_exist: bool = True, create_dir: bool = True, path_suffix: str | None = None
```

Comment:
@Innixma

Reply:
Looks good to me, but maybe we also do `path = str(path)`  inside `if path:` just so we ensure we are only working with a single type (reduces chance of bugs)